### PR TITLE
Add redirect to category for product not available

### PR DESCRIPTION
**Pull request by aleeks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Add redirect to category for product not available
| Type?         | improvement
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/STAB-51
| How to test?  | (fresh install or execute sql upgrade file) > Go to product form, apply category redirect & test

### CODE REVIEW

File: controllers/admin/AdminProductsController.php [hunk 1]
```diff
@@ -3799,7 +3799,8 @@ public function initFormInformations($product)
         ));
         $this->object = $product;
         //$this->display = 'edit';
-        $data->assign('product_name_redirected', Product::getProductName((int)$product->id_product_redirected, null, (int)$this->context->language->id));
+        // not used anymore ?
+        // $data->assign('product_name_redirected', Product::getProductName((int)$product->id_product_redirected, null, (int)$this->context->language->id));
```
Comment (thierrymarianne) [inline controllers/admin/AdminProductsController.php]: Please don't leave commented out code.

**Comment by Hamzelot** [comment]
Hello, I`m using the programm Prestashop since today. I found a problem with the "product links" in the "admin-panel", it just dont worked, but i fixed it. The "redirect to modules"-option of "modules and service" seems to have a similar problem.

**Review by maximebiloe — APPROVED** [review]

File: src/PrestaShopBundle/Controller/Admin/CategoryController.php [hunk 2]
```diff
@@ -84,4 +84,22 @@ public function addSimpleCategoryFormAction(Request $request)
...
+
+    /**
+     * Get Categories formatted like ajax_product_file.php
+     *
+     * @param $limit
+     * @param Request $request
+     * @return JsonResponse
+     */
+    public function getAjaxCategoriesAction($limit, Request $request)
+    {
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/CategoryController.php]: You should limit access to this action only its from AJAX request

Comment (aleeks) [inline src/PrestaShopBundle/Controller/Admin/CategoryController.php]: Done, thank you.

**Comment by maximebiloe** [comment]
Thanks @aleeks

**Pull request merged**